### PR TITLE
Minor fix and computational improvments

### PR DESCRIPTION
**Pull request by jcarpent**

### CODE REVIEW

File: src/narrowphase/details.h [hunk 1]
```diff
@@ -261,15 +261,15 @@ namespace fcl {
...
+      const Vec3f & o2 = tf2.getTranslation();
       Vec3f diff = o1 - o2;
       FCL_REAL len = diff.norm();
       normal = -diff/len;
       dist = len - s1.radius - s2.radius;
 
-      p1 = o1 - diff * (s1.radius / len);
-      p2 = o2 + diff * (s2.radius / len);
+      p1.noalias() = o1 - normal * s1.radius;
+      p2.noalias() = o2 + normal * s2.radius;
```
Comment (jmirabel) [inline src/narrowphase/details.h]: If `normal = - diff / len`, then, shouldn't the sign before normal change ?

Comment (jmirabel) [reply src/narrowphase/details.h]: The normal points from s1 to s2 so the sign should be changed.

Comment (jcarpent) [reply src/narrowphase/details.h]: You're right. I have done the fix.

**Comment by jcarpent** [comment]
@jmirabel You can merge when you want.

**Pull request merged**